modules/post/multi/manage/shell_to_meterpreter.rb

```diff
@@ -78,21 +82,28 @@ def run
     case session.platform
     when 'windows', 'win'
       platform = 'windows'
-      payload_name = 'windows/meterpreter/reverse_tcp'
       lplat = [Msf::Platform::Windows]
-      larch = [ARCH_X86]
-      psh_arch = 'x86'
-      vprint_status("Platform: Windows")
+      arch = cmd_exec('wmic os get osarchitecture')
+      if arch =~ /64-bit/m
+        payload_name = 'windows/x64/meterpreter/reverse_tcp'
+        larch = [ARCH_X64]
+        psh_arch = 'x64'
+      else
+        payload_name = 'windows/meterpreter/reverse_tcp'
```

Comment:
What about ARM versions of Windows? Should we be taking this into account here vs just going to an x86 payload if the target is not x64?

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/15864/commits/1b9f9f062076fa96889efac124027c75ac7c3dd2